### ISSUE TITLE
`delete` operator inconsistent with JavaScript standards

### ISSUE THREAD

**Issue by EthanRBrown** [automerge/automerge#97]
Inside an Automerge `change` callback, the `delete` operator will throw an error if the property being deleted doesn't exist:

```javascript
doc = Automerge.change(doc, '', doc => {
  delete doc.container.nonExistentProp    // throws error
})
```

This is inconsistent with JavaScript's `delete` operator which simply returns `true` if the property was successfully deleted OR it didn't exist in the first place (it only returns `false` if the property exists and couldn't be deleted, such as from a frozen object -- in any rate, it never throws a runtime error).

I believe the API behavior of Automerge should be consistent with JavaScript whenever feasible, and unless there's a good reason for this behavior, I recommend it be changed (would be happy to make that fix myself).

**Comment by ept** [comment]
Hi @EthanRBrown, I agree Automerge's behaviour should match that of JS. A pull request would be very welcome.

**Comment by EthanRBrown** [comment]
Closing - fixed by #100
